stan/math/prim/fun/grad_inc_beta.hpp

```diff
@@ -31,8 +31,9 @@ inline void grad_inc_beta(double& g1, double& g2, double a, double b,
   double dF1 = 0;
   double dF2 = 0;
   double dF3 = 0;
+  double dFz = 0;
   if (C) {
-    grad_2F1(dF1, dF2, dF3, a + b, 1.0, a + 1, z);
+    grad_2F1(dF1, dF2, dF3, dFz, a + b, 1.0, a + 1, z);
```

Comment:
[optional] instead of in/out parameters you could do something like 

```suggestion
    std::forward_as_tuple(dF1, dF2, dF3, dFz) = grad_2F1(a + b, 1.0, a + 1, z);
```

https://godbolt.org/z/4GPqjeYYf